### PR TITLE
686 add skip to topbottom links to chat

### PR DESCRIPTION
**Pull request by dhinrichs-scottlogic**
## Description
Added two skip links, which are visually hidden until you tab to them. 
Skip link 1 is at the beginning of the chatbox, which brings you to the chat input.
Skip link 2 is just above the chat input and brings you to skip link 1. 

## Screenshots
Skip link 1: 
image
Skip link 2:
image

GIVEN The the keyboard focus is on the chat input
WHEN the user tabs to the previous element (shift + tab)
THEN the "skip to top of chat" skip link shows up

WHEN the user activates that link (enter)
THEN the keyboard focus jumps to the "skip to chat input" skip link at the top of the chatbox

WHEN the user activates that link (enter)
THEN the keyboard focus jumps to the chat input

WHEN the keyboard focus is not on the skip links
THEN the skip links are not visible

## Checklist

Have you done the following?

- [x] Linked the relevant Issue
- [ ] Added tests
- [x] Ensured the workflow steps are passing

### CODE REVIEW

**Review by AAloshine-scottlogic — COMMENTED** [review]
just a small clarification, but overall looks good!

**Comment by AAloshine-scottlogic** [comment]
Also, maybe the skip to top of chat might look better below the text box?

File: frontend/src/components/ChatBox/ChatBox.tsx [hunk 1]
```diff
@@ -201,9 +201,19 @@ function ChatBox({
...
+				className="skip-to-bottom skip-link "
+				id="skip-to-bottom"
+				href="#chat-box-input"
+			>
+				<span aria-hidden>&#129035;&nbsp;</span>skip to chat input
+			</a>
 			<ChatBoxFeed messages={messages} />
 			<div className="footer">
+				<a className="skip-to-top skip-link " href="#skip-to-bottom">
+					<span aria-hidden>&#129033;&nbsp;</span>skip to top of chat
```
[the remark this answers is not in the document]
Comment (chriswilty) [reply frontend/src/components/ChatBox/ChatBox.tsx:215]: Those'll be the arrow icons, as unicode escape codes. There are a heap of arrows available, impossible to choose!

https://www.w3schools.com/charsets/ref_utf_arrows.asp

**Review by chriswilty — COMMENTED** [review]
Lovely stuff! Just the things we chatted about earlier.

To my eyes, the positioning of skip-to-top is fine, as it sits nicely in the gap between chat input and chats panel.

**Comment by dhinrichs-scottlogic** [comment]
> Also, maybe the skip to top of chat might look better below the text box? it seems pretty abrupt the way it is now? maybe some padding above/below
@AAloshine-scottlogic 

- The skip to top link below the chat input seems counterintuitive to me. Tab order is linear, so I don't want to burden users with having to remember that they need to go forwards in order to find the link to go backwards. 

That being said, if you want to wireframe your ideas and talk it through more, I'm definitely up for that, always keen to explore creative solutions :) 

- To make the skip links showing up less jarring, I'll look into adding some transition effects to blend the links in more slowly.

**Comment by chriswilty** [comment]
@dhinrichs-scottlogic A transition effect could be ok, however, you'd probably need to include `opacity` on your skip links, and transition that property.

**Comment by dhinrichs-scottlogic** [comment]
> @dhinrichs-scottlogic A transition on opacity could work in theory, but as you are setting width from 1px to 100% you won't see the fade-out, only the fade-in (or vice versa)[...]

As discussed, no fade-out, used opacity, and set the starting opacity to 0.1 instead of 0 to avoid the element being hidden from assistive tech

**Pull request merged**